edge-util/src/Microsoft.Azure.Devices.Edge.Storage.RocksDb/disk/StorageSpaceChecker.cs

```diff
@@ -0,0 +1,149 @@
+            return diskSpaceChecker;
+        }
+
+        public void SetThresholdPercentage(int thresholdPercentage)
+        {
+            lock (this.updateLock)
+            {
+                this.drive.ForEach(d =>
+                {
+                    Events.SetMaxPercentageUsage(thresholdPercentage, d);
+                    this.inner = new ThresholdStorageSpaceChecker(d, thresholdPercentage, this.checkFrequency, Events.Log);
+                });
+            }
+        }
+
+        public void SetMaxDiskUsageSize(long maxDiskUsageBytes)
+        {
+            lock (this.updateLock)
+            {
+                Events.SetMaxSizeDiskSpaceUsage(maxDiskUsageBytes, this.storageFolder);
+                this.inner = new FixedSizeSpaceChecker(this.storageFolder, maxDiskUsageBytes, this.checkFrequency, Events.Log);
+            }
+        }
+
+        internal static Option<DriveInfo> GetMatchingDrive(string storageFolder)
+        {
+            DriveInfo match = null;
+            var drives = new DriveInfo[0];
+            try
+            {
+                drives = DriveInfo.GetDrives();
+                int segmentCount = -1;
+                foreach (DriveInfo drive in drives)
+                {
+                    if (drive.IsReady)
+                    {
+                        if (storageFolder.StartsWith(drive.Name, StringComparison.OrdinalIgnoreCase))
+                        {
+                            string[] segments = drive.Name.Split(new[] { Path.DirectorySeparatorChar }, StringSplitOptions.RemoveEmptyEntries);
+                            if (segments.Length > segmentCount)
+                            {
+                                segmentCount = segments.Length;
+                                match = drive;
+                            }
+                        }
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                Events.ErrorGettingMatchingDrive(storageFolder, e);
+            }
+
+            if (match == null)
+            {
+                Events.NoMatchingDriveFound(drives, storageFolder);
+                return Option.None<DriveInfo>();
+            }
+
+            Events.FoundDrive(storageFolder, match);
+            return Option.Maybe(match);
+        }
+
+        static class Events
+        {
+            public static readonly ILogger Log = Logger.Factory.CreateLogger<StorageSpaceChecker>();
+            const int IdStart = 50000;
+
+            enum EventIds
+            {
+                Created = IdStart,
+                SetMaxSizeDiskSpaceUsage,
+                SetMaxPercentageUsage,
+                FoundDrive,
+                NoMatchingDriveFound,
+                ErrorGettingMatchingDrive
+            }
+
+            public static void Created(string storageFolder)
+            {
+                Log.LogInformation((int)EventIds.Created, $"Created disk space usage checker for folder {storageFolder}");
```

Comment:
>disk [](start = 68, length = 4)

nit: maybe change references to the word 'disk' from the logs to 'storage' as well?